### DISCUSSION TITLE
Pass a 2D vector to a parameter

### DISCUSSION THREAD

**Discussion by xyyrly** [pybamm-team/PyBaMM#3098] [Q&A]
Dear Team:

I am trying to process interfacial current density calculated by DFN model to my thermal model as a parameter, which should be a 2D spatio-temporal vector. I found an example like this: 
current_interpolant = pybamm.Interpolant(drive_cycle[:, 0], drive_cycle[:, 1], pybamm.t)
param["Current function [A]"] = current_interpolant  

But how does pybamm.Interpolant work on 2D? And how can I set the spatial positions to interpolate?(can I define and use a SpatialVariable like "x_p"?） 

Thank you for your time~

**Comment by valentinsulzer** [comment]
See https://github.com/pybamm-team/PyBaMM/blob/develop/tests/unit/test_expression_tree/test_interpolant.py#L93-L108

**Reply by xyyrly** [reply]
Thank you very much@tinosulzer
But I am still confused about how to set var1 and var2. As my understand, one of them I can use pybamm.t for the discretised time vector, but how to set the space vector? 
It looks not correct when I set as: 
x = (Xn[:,0],np.linspace(0, 1, 10))#shape(20,10)
z = J_V#shape(20,10)
x_n = pybamm.SpatialVariable("x_n", domain=["negative electrode"], coord_sys="cartesian")
param = pybamm.ParameterValues(
    {
      "Negative electrode volumetric interfacial current density [A.m-3]": pybamm.Interpolant(x, z, (x_n, pybamm.t), interpolator="linear"), 
    }
)
